design/014/mTLS.md

```diff
@@ -0,0 +1,421 @@
+# mTLS & Service Discovery for Out-of-Process Providers
+
+## Problem Description
+
+ESO Core needs to establish a secure and authenticated communication channel with out-of-process providers. It has to be encrypted in transit because we transmit secrets over the network. It must to be authenticated, otherwise a malicious actor could call GetSecret() to retrieve secrets and take advantage over the out-of-process provider's service account to read secrets or other resources within the cluster.
```

Comment:
I would argue that a lot of users of ESO will also have an existing mTLS solution as part of their service mesh, so this should be optional.

Reply:
to add more chaos juice here, some service mesh setups out there depend on external-secrets to run :laughing:

Reply:
IMO: Service Meshes are supposed to be transparent, there's nothing we need to take into consideration design-wise.